spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubChannelAdaptersIntegrationTests.java

```diff
@@ -160,18 +160,18 @@ public void sendAndReceiveMessageManualAck() {
 		this.inboundChannelAdapter.setAckMode(AckMode.MANUAL);
 		this.inputChannel.send(MessageBuilder.withPayload("I am a message.").build());
 
-		Message<?> message = this.channel.receive(1000);
+		Message<?> message = this.channel.receive(5000);
 		assertThat(message).isNotNull();
 		AckReplyConsumer acker =
 				(AckReplyConsumer) message.getHeaders().get(GcpHeaders.ACKNOWLEDGEMENT);
 		assertThat(acker).isNotNull();
 		acker.nack();
-		message = this.channel.receive(1000);
+		message = this.channel.receive(5000);
 		assertThat(message).isNotNull();
 		acker = (AckReplyConsumer) message.getHeaders().get(GcpHeaders.ACKNOWLEDGEMENT);
 		assertThat(acker).isNotNull();
 		acker.ack();
-		message = this.channel.receive(1000);
+		message = this.channel.receive(5000);
 		assertThat(message).isNull();
```

Comment:
I still think that wait here for nothing for 5 seconds is too expensive. Exactly for this `isNull()` use-case.
I think what you are waiting there is just for first message - when we warm up. All subsequent messages should be pulled from the topic pretty quickly. Otherwise something wrong with the configuration and maybe some `timeout` option should be applied.